src/v2/SuperstateToken.sol

```diff
@@ -0,0 +1,463 @@
+
+    /// @dev Event emitted when tokens are minted
+    event Mint(address indexed minter, address indexed to, uint256 amount);
+
+    /// @dev Event emitted when tokens are burned
+    event Burn(address indexed burner, address indexed from, uint256 amount);
+
+    /// @dev Emitted when the accounting pause is triggered by `admin`.
+    event AccountingPaused(address admin);
+
+    /// @dev Emitted when the accounting pause is lifted by `admin`.
+    event AccountingUnpaused(address admin);
+
+    /// @dev Thrown when a request is not sent by the authorized admin
+    error Unauthorized();
+
+    /// @dev Thrown when an address does not have sufficient permissions, as dictated by the AllowList
+    error InsufficientPermissions();
+
+    /// @dev Thrown when an address does not have a sufficient balance of unencumbered tokens
+    error InsufficientAvailableBalance();
+
+    /// @dev Thrown when the amount of tokens to spend or release exceeds the amount encumbered to the taker
+    error InsufficientEncumbrance();
+
+    /// @dev Thrown when the current timestamp has surpassed the expiration time for a signature
+    error SignatureExpired();
+
+    /// @dev Thrown if the signature has an S value that is in the upper half order.
+    error InvalidSignatureS();
+
+    /// @dev Thrown if the signature is invalid or its signer does not match the expected singer
+    error BadSignatory();
+
+    /// @dev Thrown if accounting pause is already on
+    error AccountingIsPaused();
+
+    /// @dev Thrown if accounting pause is already off
+    error AccountingIsNotPaused();
+
+    /// @dev Thrown if an address tries to encumber tokens to itself
+    error SelfEncumberNotAllowed();
+
+    /// @dev Thrown if array length arguments aren't equal
+    error InvalidArgumentLengths();
+
+    /**
+     * @notice Construct a new ERC20 token instance with the given admin and AllowList
+     * @param _allowList Address of the AllowList contract to use for permission checking
+     * @dev Disables initialization on the implementation contract
+     */
+    constructor(address _existingAdmin, AllowList _allowList) {
+        admin = _existingAdmin;
+        allowList = _allowList;
+        _disableInitializers();
+    }
+
+    /**
+     * @notice Initialize the contract
+     * @param _name The token name
+     * @param _symbol The token symbol
+     */
+    function initialize(string calldata _name, string calldata _symbol) public initializer {
+        __ERC20_init(_name, _symbol);
+        __Pausable_init();
+    }
+
+    // initialize for v2
+    function initializeV2() reinitializer(2) public {
+        // Last usage of `admin` variable here.
+        // After this call, owner() is the source of truth for authorization.
+        if (msg.sender != admin) revert Unauthorized();
+        __Ownable2Step_init();
+    }
+
+    function _requireAuthorized() internal view {
+        if (msg.sender != owner()) revert Unauthorized();
```

Comment:
big change here.

No longer user `admin` and am using `owner()`, which needs to be initialized in the proxy state by calling `initializeV2()`

Reply:
Why not use `_checkOwner` on Ownable

Reply:
Addressed in
- https://github.com/superstateinc/ustb/pull/56/commits/4e52d672532d51249bce9677ae1801a91404eda8
- https://github.com/superstateinc/ustb/pull/56/commits/863dda0005df872dec9136d026df07f918dbb2e5